packages/runtime/src/next/components/MakeswiftSlot.tsx

```diff
@@ -0,0 +1,35 @@
+'use client'
+
+import { createContext, ReactNode } from "react";
+import { MakeswiftComponentSnapshot } from "../client";
+import { MakeswiftComponentType } from "../../components";
+import { MakeswiftComponent } from "./MakeswiftComponent";
+
+
+type Props = {
+  label: string
+  snapshot: MakeswiftComponentSnapshot
+  fallback?: ReactNode
+}
+
+export const PropsContext = createContext<{fallback: ReactNode}>({fallback: null});
+
+export const PropsContextProvider = ({ value, children }: { value: {fallback: ReactNode}, children: ReactNode }) => (
+  <PropsContext.Provider value={value}>{children}</PropsContext.Provider>
+);
+
+export const MakeswiftSlot = ({ label, snapshot, fallback }: Props) => {
+  const componentType = MakeswiftComponentType.Slot
+
+  return (
+    <PropsContextProvider value={{fallback}}>
```

Comment:
This will create a new object on every render. We can use `useMemo` to avoid that.